app/models/report.rb

```diff
@@ -4,21 +4,31 @@ class Report < ApplicationRecord
   validates :user_id, presence: true
   validates :item_id, presence: true
   validates :item_type, presence: true, inclusion: {
-    in: %w(Post Comment), message: "Type should match `Post` or `Comment`!"}
+    in: %w[Post Comment], message: "Type should match `Post` or `Comment`!"
+  }
   validates :text, presence: true
 
-  validate :entry_does_not_exist, :on => :create
-  validate :post_or_comment_does_exist, :on => :create
+  validate :entry_does_not_exist, on: :create
+  validate :post_or_comment_does_exist, on: :create
 
   belongs_to :user
   belongs_to :post, optional: true
   belongs_to :comment, optional: true
   belongs_to :item, polymorphic: true
+  delegate :author, to: :item
```

Comment:
Adding the author of the item as `author` here is very confusing, as `report.author` then looks like this is the author if the report, please add at least a `prefix: true` here.

Also, as the item might get deleted, it would probably be smarter to instead use the new column which stores the author of the reported item even if the item itself gets deleted.

Reply:
Hm, if the item is deleted the author is also not every interesting.  I'll take the prefix.

Reply:
Hm 2: There exists already a 'reported_author' in the report model. This is already a better name for what it means. I drop the delegate and use the existing code.